packages/cloud_firestore/ios/Classes/CloudFirestorePlugin.m

```diff
@@ -143,6 +143,9 @@ - (void)writeValue:(id)value {
     NSString *documentPath = [documentReference path];
     [self writeByte:DOCUMENT_REFERENCE];
     [self writeUTF8:documentPath];
+  } else if ([value isKindOfClass:[NSData class]]) {
+    NSData *blob = value;
+    [super writeValue:blob.bytes];
```

Comment:
This won't work. But there is a `writeData` method that takes an `NSData` directly.

Reply:
https://docs.flutter.io/objcdoc/Classes/FlutterStandardWriter.html#/c:objc(cs)FlutterStandardWriter(im)writeData:

Reply:
That would require you to write a type discriminator byte first though (also from the Android side). Otherwise, you can wrap your `NSData` in a [FlutterStandardTypedData](https://docs.flutter.io/objcdoc/Classes/FlutterStandardTypedData.html) which can be directly given as argument to `writeValue`.

Reply:
Thank you very much for your help and hints. I will go with the explicit type discriminator byte to keep it more explicit and also more consistent with the existing implementation.